versioned_docs/version-1.5(WIP)/gwScripts.md

```diff
@@ -0,0 +1,204 @@
+
+### Challenge Lock
+
+When a Godwoken node found that an invalid state in the Rollup, it can send a `RollupEnterChallenge` action to the Rollup cell and generate a challenging cell. A challenge cell must set a challenging target in its `ChallengeLockArgs` lock args. The challenging target can be a layer-2 transaction or a withdrawal request.
+
+If the challenging unit does not get cancelled when completed, the challenger can execute the `RollupRevert` action on the Rollup cell and take stake cell sent by reverted block submitter as a reward.
+
+If the challenging target is invalid, other nodes can cancel this challenge by executing the `RollupCancelChallenge` action, and the challenging cell must be included in the tx.inputs.
+
+* The challenge lock verifies that the account script is in tx.inputs in order to verify the signature for a withdrawal target.
+* For a layer-2 transaction target, challenge lock reads the backend script code_hash from the state tree, then verifies if the backend validator script is in the tx.inputs.
+
+## Layer 2 Scripts
+
+The C scripts that located in the `c` directory are Godwoken layer-2 scripts. The layer-2 script can be executed on CKB when a challenge occurs, which means that a layer-2 script is also a valid layer-1 script, only it follows the special interface convenience required by Godwoken.
+
+Godwoken account consisted of the following fields: `(id: u32, nonce: u32, script: Script)`. The `script` field determines the script that the account used. 
+
+Layer-2 scripts exist in two types: lock and contract. If an account ID appears in `l2tx.from_id`, we will assume that the account's script is a lock, implying that the script follows the lock script interface convenience that can verify signatures(e.g. Ethereum EOA). If an account ID appears in `l2tx.to_id`, we will assume that the account's script is a contract, which means we should execute the tx when it is sent to the account(like an Ethereum contract account).
+
+A layer-2 contract script is run both on and off-chain. The unified interface is defined in the `c/gw_def.h`. The on-chain implementation is `validator_utils.h`, and the off-chain implementation is `generator_utils.h`.
+
+### ETH Account Lock
+
+ETH Account Lock is a 2-layer lock script that verifies the layer 2 account signature.
+
+### Meta Contract
+
+Meta contract is a layer-2 contract script. A built-in layer-2 account allows creating another account by sending a tx to the account. Meta contract args is `MetaContractArgs`, the built-in contract id is `0`.
+
+### sUDT Contract
+
+sUDT contract is a layer-2 contract script that keeps a consistent mapping to the layer-1 sUDT. The layer-1 sUDT script hash is equal to `account.script.args`. When the user deposits a new type of sUDT, Godwoken will create a new corresponding sUDT account. This contract args is `SUDTArgs`, the built-in CKB Simple UDT contract id is `1`.
```

Comment:
-  the built-in CKB Simple UDT contract id is `1`. --> and the built-in CKB Simple UDT contract id is `1`.